### DISCUSSION TITLE
Selecting a primary color from existing color plate

### DISCUSSION THREAD

**Discussion by berkayyildiz** [tailwindlabs/tailwindcss#8524] [Help]
Hello, as an example I have a button, it has bg-primary-200 and text-primary-800 classes. I want to set that primary from existing tailwind plate. For example I want to add primary=red attribute and make component use red-200 and red-800. I can make this manually in scss but its a lot of work for every color even with mixins because it is not possible to iterate color names due to compile time generation of classes. Do you know a way to achieve this?

**Comment by crswll** [comment]
You can use `tailwindcss/colors`.

```js
const colors = require('tailwindcss/colors')

module.exports = {
  theme: {
    extend: {
      colors: {
        primary: colors.pink
      },
    },
  },
}
```

Then you can do `bg-primary-200` and `text-primary-700` and so on.
